scripts/install/calibre-web.sh

```diff
@@ -0,0 +1,124 @@
+		python3 -m venv /opt/.venv/calibre-web
+	fi
+	echo_progress_done
+
+	echo_progress_start "Downloading source code archive"
+	dlurl=$(curl -s https://api.github.com/repos/janeczku/calibre-web/releases/latest | grep "browser_download_url" | head -1 | cut -d\" -f 4)
+	# shellcheck disable=SC2181
+	if [[ $? != 0 ]]; then
+		echo_error "Failed to query github"
+		exit 1
+	fi
+
+	if ! wget "${dlurl}" -O /tmp/calibreweb.zip >> $log 2>&1; then
+		echo_error "Failed to download binary"
+		exit 1
+	fi
+	echo_progress_done
+
+	echo_progress_start "Extracting archive"
+	unzip /tmp/calibreweb.zip -d /tmp/calibrewebdir >> $log 2>&1
+	subdir=$(ls /tmp/calibrewebdir)
+	mv /tmp/calibrewebdir/"$subdir" $calibrewebdir
+	echo_progress_done
+
+	echo_progress_start "Creating users and setting permissions"
+	useradd $calibrewebuser --system -d "$calibrewebdir" >> $log 2>&1
+	if [[ -z $calibredbuser ]]; then
+		calibredbuser=$(cut -d: -f1 < /root/.master.info)
+	fi
+	usermod -a -G "${calibredbuser}" $calibrewebuser >> $log 2>&1
+	chown -R $calibrewebuser:$calibrewebuser $calibrewebdir
+	chown -R ${calibrewebuser}: /opt/.venv/calibre-web
+	echo_progress_done
+
+	echo_progress_start "Installing python dependencies"
+	sudo -u ${calibrewebuser} bash -c "/opt/.venv/calibre-web/bin/pip3 install -r $calibrewebdir/requirements.txt" >> $log 2>&1
+	#fuck ldap. all my homies hate ldap
+	sed '/ldap/Id' -i $calibrewebdir/optional-requirements.txt
+	sudo -u ${calibrewebuser} bash -c "/opt/.venv/calibre-web/bin/pip3 install -r $calibrewebdir/optional-requirements.txt" >> $log 2>&1
+	echo_progress_done
+}
+
+_install_kepubify() {
+	echo_progress_start "Installing kepubify"
+	wget "https://github.com/pgaskin/kepubify/releases/download/v3.1.2/kepubify-linux-64bit" -O /tmp/kepubify >> $log 2>&1
+	chmod a+x /tmp/kepubify
+	mv /tmp/kepubify /usr/local/bin/kepubify
+	#TODO and figure out if it's needed for all cases or not
+	echo_progress_done
+}
+
+_nginx_calibreweb() {
+	if [[ -f /install/.nginx.lock ]]; then
+		echo_progress_start "Setting up nginx conf"
+		bash /etc/swizzin/scripts/nginx/calibre-web.sh
+		echo_progress_done
+	fi
+}
+
+_systemd_calibreweb() {
+	echo_progress_start "Creating and enabling systemd services"
+	cat > /etc/systemd/system/calibre-web.service << EOF
+[Unit]
+Description=Calibre-Web
+
+[Service]
+User=$calibrewebuser
+Type=simple
+ExecStart=/opt/.venv/calibre-web/bin/python3 $calibrewebdir/cps.py
+WorkingDirectory=$calibrewebdir
+Environment=PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin:/opt/.venv/calibre-web/bin
+
+[Install]
+WantedBy=multi-user.target
+EOF
+	systemctl daemon-reload -q 2>&1 | tee -a $log
+	systemctl enable -q --now calibre-web.service 2>&1 | tee -a $log
+	echo_progress_done
+}
+
+_postinstall_calibreweb() {
+	echo_warn "Please continue the configuration and installation of Calibre-web through your browser."
+	hostname="<domain/IP>" #figure this one out dynamically maybe?
```

Comment:
We have the IP command:

```
ip=$(ip route get 1 | sed -n 's/^.*src \([0-9.]*\) .*$/\1/p')
```

Hostname is tough because I'd warrant a guess that no one is setting up their hosts correctly to use something like `hostname -f`

Reply:
I'm gonna just do away with it completely and change the wording